src/main/java/org/humanistika/oxygen/tei/completer/TeiCompleter.java

```diff
@@ -130,19 +147,24 @@ protected final AutoCompleteSuggestions<AutoComplete> getAutoCompleteSuggestions
         return null;
     }
 
-    protected List<CIValue> requestAutoComplete(final AutoComplete autoComplete, final String selection, @Nullable final String dependent) {
+    public List<CIValue> requestAutoComplete(final AutoComplete autoComplete, final String selection, @Nullable final String dependent) {
         final Authentication.AuthenticationType authenticationType = autoComplete.getRequestInfo().getAuthentication() == null ? null : autoComplete.getRequestInfo().getAuthentication().getAuthenticationType();
-        final Suggestions suggestions = getClient(authenticationType).getSuggestions(autoComplete.getRequestInfo(), selection, dependent, autoComplete.getResponseAction());
-        final List<CIValue> results = new ArrayList<>();
-        for(final Suggestion suggestion : suggestions.getSuggestion()) {
-            results.add(new CIValue(suggestion.getValue(), suggestion.getDescription()));
+        //TODO USE a constant for this value
+        if(selection.length() > 3 && dependent.length() > 3) {
+            final Suggestions suggestions = getClient(authenticationType).getSuggestions(autoComplete.getRequestInfo(), selection, dependent, autoComplete.getResponseAction());
+            final List<CIValue> results = new ArrayList<>();
+            for(final Suggestion suggestion : suggestions.getSuggestion()) {
+                results.add(new CIValue(suggestion.getValue(), suggestion.getDescription()));
+            }
+            return results;
         }
-        return results;
+        return new ArrayList<>();
```

Comment:
Do you mean to return an empty `ArrayList` here?

If so, you should instead use `Collections.emptyList()` here.